test/e2e/performanceprofile/functests/5_latency_testing/latency_testing.go

```diff
@@ -121,17 +122,27 @@ var _ = table.DescribeTable("Test latency measurement tools tests", func(testGro
 				testlog.Error(matchErr.Error())
 			}
 			if ok {
-				var commandRegex string
-				if test.toolToTest == oslat {
-					commandRegex = fmt.Sprintf("Running the oslat command with arguments .*--duration %s", test.testRuntime)
+				//verify the command is executed with the expected args
+				//this lists of args depend on the ones the latency tool runners adds to tool command in cnf-features-deploy.
+				var passedArgs []string
+				switch test.toolToTest {
+				case oslat:
+					passedArgs = []string{"--duration " + test.testRuntime, "--rtprio ", "--cpu-list ", "--cpu-main-thread "}
+				case cyclictest:
+					passedArgs = []string{"--duration " + test.testRuntime, "--priority 95", "--threads ", "--affinity ", "--histogram ", "--interval ", "--mlockall ", "--quiet"}
+				case hwlatdetect:
+					thr := test.testMaxLatency
+					if test.hwlatdetectMaxLatency != "" {
+						thr = test.hwlatdetectMaxLatency
+					}
+					passedArgs = []string{"--duration " + test.testRuntime, "--threshold " + thr, "--hardlimit " + thr, "--window ", "--width "}
+				default:
+					testlog.Error("the tool to test was not set")
 				}
-				if test.toolToTest == cyclictest {
-					commandRegex = fmt.Sprintf("running the cyclictest command with arguments .*-D %s", test.testRuntime)
-				}
-				if test.toolToTest == hwlatdetect {
-					commandRegex = fmt.Sprintf("running the hwlatdetect command with arguments .*--duration %s", test.testRuntime)
+
+				for _, argument := range passedArgs {
+					Expect(strings.Contains(string(output), argument)).To(BeTrue(), "The tool command didn't pass the argument %q", argument)
```

Comment:
please check the gomega matchers `ContainElement` and `ContainSubstring`

Reply:
That sounds a good suggestion but it seems that ContainElement matcher expects array/ slice/map:
"Actual must be an array, slice or map"
https://onsi.github.io/gomega/#containelementselement-interface

Reply:
fair enough, feel free to ignore